File: app/backend/models/tipo_alerta_model.py
from sqlalchemy import Column, Integer, String, SmallInteger, DateTime
from backend.database.connection import Base

class TipoAlerta(Base):
    __tablename__ = "TipoAlerta"

    IdTipoAlerta = Column(Integer, primary_key=True, index=True, autoincrement=True)
    Descripcion = Column(String(45), nullable=False)
    IdArea = Column(SmallInteger, nullable=True)
    Asunto = Column(String(50), nullable=True)
    Mensaje = Column(String(5000), nullable=True)
    Obs = Column(String(5000), nullable=True)
    AudFecha = Column(DateTime, nullable=True)
    AudUsuario = Column(SmallInteger, nullable=True)

File: app/frontend/src/app/features/tipos-alerta/models/tipo-alerta.model.ts
export interface TipoAlerta {
  IdTipoAlerta: number;
  Descripcion: string;
  IdArea?: number;
  Asunto?: string;
  Mensaje?: string;
  Obs?: string;
  AudFecha?: string;
  AudUsuario?: number;
}

export interface TipoAlertaCreate {
  Descripcion: string;
  IdArea?: number;
  Asunto?: string;
  Mensaje?: string;
  Obs?: string;
}

export interface Area {
  IdArea: number;
  Descripcion: string;
  AudFecha?: string;
  AudUsuario?: number;
}